File: src/lib/software/teracopy.py
from src.lib.software import BaseSoftware

class TeraCopy(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.download_name = 'teracopy.exe'

    def resolve_download_url(self):
        self.url_resolved.emit('https://www.codesector.com/files/teracopy.exe')

File: packages/app/src/services/software/definitions/teracopy.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'teracopy';
	public name = 'TeraCopy';
	public category = [SoftwareCategory.FileManagement, SoftwareCategory.Utility];
	public downloadName = 'teracopy.exe';
	public icon = 'teracopy.png';
	public homepage = 'https://codesector.com/teracopy';

	public async resolveDownloadUrl() {
		return ok('https://codesector.com/files/teracopy.exe');
	}
}
